Source/WebCore/platform/audio/gstreamer/AudioDecoderGStreamer.cpp

```diff
@@ -334,16 +335,14 @@ void GStreamerInternalAudioDecoder::decode(std::span<const uint8_t> frameData, b
             m_harness->processOutputSamples();
         else
             m_outputCallback(makeUnexpected("Decode error"_s));
-
-        callback({ });
     });
+    return AudioDecoder::DecodePromise::resolve();
```

Comment:
this is incorrect, we are returning a resolved promise immediately before the work as completed.
We have to create a NativePromise::Producer pass it to the lambda and resolve it where `callback` was called.

Reply:
This is inside GStreamerInternalAudioDecoder not GStreamerAudioDecoder  so the work has completed.